dluhc-component-library/src/components/timetablePage/TimetablePage.tsx

```diff
@@ -2,42 +2,54 @@ import { useEffect, useState } from "preact/hooks";
 import csvToJson from "csvtojson";
 import { loadCSV, loadJson } from "../../utils";
 import Timetable from "./Timetable";
-import { TimetableStage } from "./types";
+import { TimetableHeader, TimetableStage } from "./types";
 import AccordionDropdown from "./AccordionDropdown";
 
 interface TimetablePageProps {
   filepath: string;
+  headersFilepath: string;
 }
 
-const TimetablePage = ({ filepath }: TimetablePageProps) => {
+const TimetablePage = ({ filepath, headersFilepath }: TimetablePageProps) => {
   const [timetableData, setTimetableData] = useState<TimetableStage[]>();
+  const [timetableHeaderData, setTimetableHeaderData] =
+    useState<TimetableHeader>();
 
-  useEffect(() => {
+  async function loadData() {
     if (/.csv$/.test(filepath)) {
-      loadCSV(filepath).then((data) => {
+      await loadCSV(filepath).then((data) => {
         csvToJson()
           .fromString(data)
           .then((jsonObj) => setTimetableData(jsonObj as TimetableStage[]));
       });
     } else if (/.json$/.test(filepath)) {
-      loadJson(filepath).then((data) => {
+      await loadJson(filepath).then((data) => {
         setTimetableData(data);
       });
     }
-  }, [setTimetableData, filepath]);
+
+    await loadCSV(headersFilepath).then((data) => {
+      csvToJson()
+        .fromString(data)
+        .then((jsonObj) => {
+          setTimetableHeaderData(jsonObj[0] as TimetableHeader);
+        });
+    });
+  }
+
+  useEffect(() => {
+    loadData();
```

Comment:
`Non-blocker` Not ideal that we need to load both of files again when there is any change to either but for the POC lets leave it like this as they are not props that change often